examples/keras_io/vision/image_classification_using_global_context_vision_transformer.py

```diff
@@ -0,0 +1,1182 @@
+module. As we can see from the `call` method,
+1. This module first **pads** input
+2. Then uses **convolutions** to extract patches with embeddings.
+3. Finally, uses `ReduceSize` module to first extract features with **convolution** but
+neither reduces spatial dimension nor increases spatial dimension.
+4. One important point to notice, unlike **ViT** or **SwinTransformer**, **GCViT**
+creates **overlapping patches**. We can notice that from the code,
+`Conv2D(self.embed_dim, kernel_size=3, strides=2, name='proj')`. If we wanted
+**non-overlapping** patches then we would've used the same `kernel_size` and `stride`.
+5. This module reduces the spatial dimension of input by $4x$.
+> **Summary:** image $\rightarrow$ padding $\rightarrow$ convolution $\rightarrow$
+(feature_extract + downsample)
+"""
+
+
+class PatchEmbed(keras.layers.Layer):
+    """Patch embedding block.
+
+    Args:
+        embed_dim: feature size dimension.
+    """
+
+    def __init__(self, embed_dim, **kwargs):
+        super().__init__(**kwargs)
+        self.embed_dim = embed_dim
+
+    def build(self, input_shape):
+        self.pad = keras.layers.ZeroPadding2D(1, name="pad")
+        self.proj = keras.layers.Conv2D(self.embed_dim, 3, 2, name="proj")
+        self.conv_down = ReduceSize(keepdims=True, name="conv_down")
+
+    def call(self, inputs, **kwargs):
+        x = self.pad(inputs)
+        x = self.proj(x)
+        x = self.conv_down(x)
+        return x
+
+
+"""
+### Global Token Gen.
+
+> **Notes:** It is one of the two **CNN** modules that is used to imppose inductive bias.
+
+As we can see from above cell, in the `level` we have first used `to_q_global/Global
+Token Gen./FeatExtract`. Let's try to understand how it works,
+
+* This module is series of `FeatureExtract` module, according to paper we need to repeat
+this module $K$ times, where
+$$K = \log_{2}(\frac{H}{h})$$
+$$H = feature\_map\_height$$
+$$W = feature\_map\_width$$
+* `FeatExtract:` This layer is very similar to `ReduceSize` module except it uses
+**MaxPooling** module to reduce the dimension, it doesn't increse feature dimension
+(channelsie) and it doesn't uses **LayerNormalizaton**. This module is used to in
+`Generate Token Gen.` module repeatedly to generte **global tokens** for
+**global-context-attention**.
+* One important point to notice from the figure is that, **global tokens** is shared
+across the whole image which means we use only **one global window** for **all local
+tokens** in a image. This makes the computation very efficient.
+* For input feature map with shape $(B, H, W, C)$, we'll get output shape $(B, h, w, C)$.
+If we copy these global tokens for total $M$ local windows in a image where, $M = \frac{H
+\times W}{h \times w} = num\_window$, then output shape: $(B * M, h, w, C)$.
+
+> **Summary**: This module is used to `resize` the image to fit window.
+
+<img
+src="https://raw.githubusercontent.com/awsaf49/gcvit-tf/main/image/global_token_annot.png"
```

Comment:
You can just use markdown syntax for images. Not specifying the width is better.